Lib/fontTools/cffLib/__init__.py

```diff
@@ -867,7 +867,12 @@ def decompile(self):
         if self.file:
             # read data in from file. Assume position is correct.
             length = readCard16(self.file)
-            self.data = self.file.read(length)
+            # https://github.com/fonttools/fonttools/issues/3673
+            if length == 65535:
+                length = 65536 * 16  # *This* out to be enough for anyone.
+                self.data = self.file.read()
```

Comment:
I assume you meant this?
```suggestion
                self.data = self.file.read(length)
```

Reply:
Did you check the failures we're getting if we unconditionally do `self.file.read()`? I couldn't immediately make sense of them, but it does make me worry that leaving the file pointer at the wrong location may have consequences elsewhere.

And since this block isn't covered by tests, we don't know, really.

Reply:
> I assume you meant this?

If so, then the file read can move out of the if/else.

Reply:
Done.

> Did you check the failures we're getting if we unconditionally do self.file.read()? I couldn't immediately make sense of them, but it does make me worry that leaving the file pointer at the wrong location may have consequences elsewhere.

You mean test failures? I don't get any.

Reply:
I think have messed up something, I can't reproduce now.